cli/src/cpp/fastdds_qos_profiles_manager_cli.cpp

```diff
@@ -0,0 +1,159 @@
+// Copyright 2023 Proyectos y Sistemas de Mantenimiento SL (eProsima).
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <iostream>
+#include <map>
+#include <string>
+
+#include <docopt/docopt.h>
+
+#include "config.h"
+#include "parser_constants.hpp"
+#include "subparsers/clear_subparser.hpp"
+#include "subparsers/help_subparser.hpp"
+#include "subparsers/print_subparser.hpp"
+#include "subparsers/query_subparser.hpp"
+#include "subparsers/set_subparser.hpp"
+#include "usages.hpp"
+
+using namespace eprosima::qosprof_cli;
+
+int main(
+        int argc,
+        char* argv[])
+{
+    std::map<std::string, docopt::value> args = docopt::docopt(
+        USAGE,
+        {argv + 1, argv + argc },
+        true,                                                       // show help if requested
+        FASTDDS_QOS_PROFILES_MANAGER_CLI_VERSION_STR,               // version string
+        true);                                                      // options first
+
+    std::string verb = args[PARSER_VERB].asString();
+    if (verb == SET_VERB)
+    {
+        // Set verb requires at least one more follow-up arguments: element being set
+        if (0 < args[PARSER_ARGS].asStringList().size())
```

Comment:
Should it be at least one?
```suggestion
        if (1 < args[PARSER_ARGS].asStringList().size())
```

Reply:
In fact this is not true. Binary properties when supported would have the following call:

```
fastddsqosprof file.xml set participant["participant_profile_name"].binary_property["Property Name 1"]
```

Consequently the number of arguments would have to be done at a lower level.

Reply:
Look at the comparison operator. `<` implies that 0 is not a valid value. Otherwise operator `<=` should be required.